src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -91,6 +93,25 @@ object Validation {
     }
   }
 
+  def fetchUriHasSupportedProtocol: Validator[FetchUri] = {
+    lazy val supportedProtocols = Set("http", "https", "ftp", "ftps", "hdfs")
```

Comment:
You are right - this will require to keep mesos/marathon capabilities in sync. I can remove this validation but failure will occur much deeper - on the mesos level